src/data/matrix/basic.lean

```diff
@@ -644,6 +644,22 @@ lemma smul_mul_vec_assoc (A : matrix m n α) (b : n → α) (a : α) :
   (a • A).mul_vec b = a • (A.mul_vec b) :=
 by { ext, apply smul_dot_product }
 
+lemma mul_vec_add (A : matrix m n α) (x y : n → α) :
+  A.mul_vec (x + y) = A.mul_vec x + A.mul_vec y :=
+by { ext, apply dot_product_add }
+
+lemma add_mul_vec (A B : matrix m n α) (x : n → α) :
+  (A + B).mul_vec x = A.mul_vec x + B.mul_vec x :=
+by { ext, apply add_dot_product }
+
+lemma vec_mul_add (A : matrix m n α) (x y : m → α) :
+  vec_mul (x + y) A = vec_mul x A + vec_mul y A :=
+by { ext, apply add_dot_product }
+
+lemma add_vec_mul (A B : matrix m n α) (x : m → α) :
+  vec_mul x (A + B) = vec_mul x A + vec_mul x B :=
+by { ext, apply dot_product_add }
```

Comment:
I'd argue that these lemma names should be swapped; in the first one, the left argument of `vec_mul` is `add`, but you put the word `add` on the right, and vice-versa.

Reply:
```suggestion
lemma add_vec_mul (A : matrix m n α) (x y : m → α) :
  vec_mul (x + y) A = vec_mul x A + vec_mul y A :=
by { ext, apply add_dot_product }

lemma vec_mul_add (A B : matrix m n α) (x : m → α) :
  vec_mul x (A + B) = vec_mul x A + vec_mul x B :=
by { ext, apply dot_product_add }
```

Reply:
I see what you mean now, will make the changes.